AudioPlayerRecorderLibrary/src/main/java/com/heavyplayer/audioplayerrecorder/fragment/AudioRecorderFragment.java

```diff
@@ -11,144 +17,147 @@
-		super.onCreate(savedInstanceState);
-
-		mStartRecorderOnBind = savedInstanceState == null;
-	}
-
-	@Override
-	public void onResume() {
-		super.onResume();
-		mAudioRecorderServiceManager.onFragmentResume();
-	}
-
-	@Override
-	public void onPause() {
-		super.onPause();
-		mAudioRecorderServiceManager.onFragmentPause();
-	}
-
-	public Dialog onCreateDialog(Bundle savedInstanceState) {
-		final Activity activity = getActivity();
-
-		final View view = LayoutInflater.from(activity).inflate(R.layout.audio_recorder, null);
-
-		return new AlertDialog.Builder(activity)
-				.setTitle(R.string.audio_recorder_recording)
-				.setView(view)
-				.setPositiveButton(R.string.audio_recorder_stop_recording, null)
-				.create();
-	}
-
-	public Uri getFileUri() {
-		if(mFileUri == null)
-			mFileUri = getArguments().getParcelable(ARG_FILE_URI);
-		return mFileUri;
-	}
-
-	@Override
-	final public void onClick(View v) {
-		final AudioRecorderService.LocalBinder binder = mAudioRecorderServiceManager.getBinder();
-		final boolean isRecording = binder != null && binder.isRecording();
-		onMicrophoneClick(v, isRecording);
-	}
-
-	protected void onMicrophoneClick(View v, boolean isRecording) {
-		final AudioRecorderService.LocalBinder binder = mAudioRecorderServiceManager.getBinder();
-		if(binder != null) {
-			if(binder.isRecording())
-				binder.stopRecorder();
-			else
-				binder.startRecorder(getFileUri());
-		}
-	}
-
-	@Override
-	public void onServiceBind(IBinder binder) {
-		registerOnAudioRecorderService((AudioRecorderService.LocalBinder)binder);
-
-		if(mStartRecorderOnBind) {
-			mStartRecorderOnBind = false;
-			((AudioRecorderService.LocalBinder)binder).startRecorder(getFileUri());
-		}
-	}
-
-	@Override
-	public void onServiceUnbind(IBinder binder) { }
-
-	protected void registerOnAudioRecorderService(AudioRecorderService.LocalBinder binder) {
-		final AudioRecorderMicrophone microphone =
-				(AudioRecorderMicrophone)getDialog().findViewById(android.R.id.input);
-		if(microphone != null)
-			microphone.setOnClickListener(this);
-
-		binder.register(microphone, this);
-	}
-
-	@Override
-	public void onServiceStop() {
-		// Purposely empty.
-	}
-
-	@Override
-	public void onStartRecorder() {
-		Activity activity = getActivity();
-		if(activity != null)
-			activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
-	}
-
-	@Override
-	public void onStartRecorderFailed(Exception e) {
-		// Purposely empty.
-	}
-
-	@Override
-	public void onStopRecorder() {
-		Activity activity = getActivity();
-		if(activity != null)
-			activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
-	}
-
-	@Override
-	public void onTimeLimitExceeded() {
-		// Purposely empty.
-	}
+        implements View.OnClickListener, ServiceManager.StateListener, AudioRecorderService.AudioRecorderStateListener {
+    public static final String TAG = AudioRecorderFragment.class.getSimpleName();
```

Comment:
Please use `AudioRecorderFragment.class.getName()` for tagging purposes. Using only the name makes the probability of clashes high, given proguard changes all class names to `a`, `b`, `c`. etc inside each package. For logging purposes `getSimpleName()` is OK.

To explain this further, we usually have these 2 for a given `Class`:
```
public static final String TAG = Class.class.getName(); // For tagging and other internal mechanisms.
public static final String LOG_TAG = Class.class.getSimpleName(); // For logging.
```
(Although, of course, we only add them when needed)